### ISSUE TITLE
UnusedVariable should not get triggered if the variable is assigned 'null'

### ISSUE THREAD

**Issue by Flowdalic** [google/error-prone#1279]
The UnusedVariable bug pattern should not get triggered if the variable is assigned 'null'.

Assigning a variable to `null` is an often used pattern to make the object the variable pointed to collectible by the garbage collector.

Consider the following (broken, because of a potential endless loop) example code
```java
private static void triggerGarbageCollection() {
    Object object = new Object();
    WeakReference<Object> weakReference = new WeakReference<>(object);
    object = null;

    do {
        System.gc();
    } while (weakReference.get() != null);
}
```

ErrorProne 2.3.3 reports:
```
> Task :example:compileJava
/example/util/MemoryLeakTestUtil.java:143: error: [UnusedVariable] The assignment to this local variable 'object' is never read.
        object = null;
               ^
    (see https://errorprone.info/bugpattern/UnusedVariable)
  Did you mean to remove this line?
1 error
```

**Comment by Stephan202** [comment]
This will be fixed in the upcoming release, see dd5d1cab88cae2c128d30b3686316acbeb1b9edf :)
